### DISCUSSION TITLE
pscale connect with service token errored with `context deadline exceeded`

### DISCUSSION THREAD

**Discussion by tuannguyen-qz** [planetscale/discussion#621] [Bug Reports]
Hello! When using service token with pscale connect, after the connection was established for about 3 seconds, it was terminated with `context deadline exceeded`.

```
pscale connect recommendations dev --service-token <retracted> --service-token-id <retracted> --org <retracted> --role reader --debug
Secure connection to database recommendations and branch dev is established!.

Local address to connect your application: 127.0.0.1:3306 (press ctrl-c to quit)
Error: deadline_exceeded: context deadline exceeded
```

I'm running on macOS Sonoma 14.3.1

**Comment by orware** [comment]
Hello @tuannguyen-qz,

I know this is separate from your other discussion created a few minutes ago, but the error here definitely does seem a bit odd considering it occurs relatively quickly.

After the `Error: deadline_exceeded: context deadline exceeded` error occurs, does the Terminal return control to the command prompt like the `pscale connect` operation terminated completely?

Do you know if you have any local MySQL instances at all that may be running on port 3306 already, or if there is any security type software installed on the machine that might attempt to close the connections?

On my end for example, port 3306 is already in use but the command should automatically have selected a new port in that case for you so this is what I got when I tried:

```
Tried address 127.0.0.1:3306, but it's already in use. Picking up a random port ...
Local address to connect your application: 127.0.0.1:49835 (press ctrl-c to quit)
```

But on the current `pscale` version of v0.186.0 I didn't encounter the same behavior you described and it looks like things were maintained past 3 seconds without issue.

We can also move this to a ticket-based conversation too if you'd prefer since it is easier to go back and forth that way if needed (https://support.planetscale.com/).

**Reply by tuannguyen-qz** [reply]
> After the Error: deadline_exceeded: context deadline exceeded error occurs, does the Terminal return control to the command prompt like the pscale connect operation terminated completely?

Yes

> Do you know if you have any local MySQL instances at all that may be running on port 3306 already, or if there is any security type software installed on the machine that might attempt to close the connections?

There is nothing running on port 3306 already. The only security type software I can think of is ZScaler but I was able to connect using `pscale auth login`.

I upgraded my `pscale` version from v0.183.0 to v0.186.0 and still got the same issue.
